llvm_passes/HipKernelArgSpiller.cpp

```diff
@@ -97,7 +97,14 @@ bool isSpecial(const Argument &Arg) {
   if (!PtrTy)
     return false;
 
-  auto *STy = dyn_cast<StructType>(PtrTy->getNonOpaquePointerElementType());
+#if LLVM_VERSION_MAJOR < 16
+    auto *PossibleSTy = Ty->getNonOpaquePointerElementType();
+#else
+    auto *PossibleSTy = Arg.hasByValAttr() ? Arg.getParamByValType()
+                                           : Arg.getType();
+#endif
+
+  auto *STy = dyn_cast<StructType>(PossibleSTy);
```

Comment:
For LLVM-17, this does not seem to cover image and sampler types coming from HipTextureLowering which are TargetExtTypes. The fix is simple: If see a TargetExtType, return true.

Reply:
good catch, fixed